### DISCUSSION TITLE
Cannot send a message larger than 4096 bytes

### DISCUSSION THREAD

**Discussion by lanner17** [gorilla/websocket#963] [Q&A]
I am not able to send a message larger that 4096 byte with `WriteMessage`. The documentation states that the write buffer size does not limit the size of the message. Also, from the code it seems `WriteMessage` is a wrapper function that calls NextWriter to "flush" messages. At the same time, I have no problem receiving messages larger than 4096 bytes. I do not really want to increase the buffer size, so any other option to handle this would be appreciated.
Thanks

**Comment by ghost** [comment]
The documentation is correct regarding the buffer size.  

Describe in more detail what happens when the application writes a message larger than the buffer size.  Does WriteMessage return an error? What is the peer?  Does the peer receive any data?   Show the code that writes the message larger than.

**Comment by lanner17** [comment]
Hi @hulkingshtick , thank you for the response. I did not see an error. Also, the peer never received any data. However, I resolved the problem by increasing the write buffer size
```
var myDefaultDialer = &websocket.Dialer{
	WriteBufferSize: 4096 * 5,
}
```
Thanks

**Reply by ghost** [reply]
A possible problem is that the server does not fully support the websocket protocol.  Specifically, the server does not handle message fragmentation.  If that is the case, then increasing the client buffer size is the only workaround for the broken server. 

The Gorilla client fragments messages when the buffer size is smaller than the message size.
